### PR TITLE
Add docs in export interface configurations for controllers.

### PR DESCRIPTION
**Pull request by destogl**
Co-authored with: Paul Gesel <paul.gesel@picknik.ai>

### CODE REVIEW

File: controller_interface/include/controller_interface/controller_interface_base.hpp [hunk 1]
```diff
@@ -72,9 +72,37 @@ class ControllerInterfaceBase : public rclcpp_lifecycle::node_interfaces::Lifecy
...
+  /// Get configuration for controller's required command interfaces.
+  /**
+   * Method used by controller manager to get configuration of controller's command interface.
+   * Each controller can use individual method to determine interface names that in simples case
+   * have the following format: `<joint>/<interface>`.
+   * The method is called only in `inactive` or `active` state, i.e., `on_configure` has to be
+   * called first.
+   * The configuration is used to check if controller can be activated and to claim interfaces from
+   * hardware.
+   * The claimed interface are populated in the
```
Comment (bmagyar) [inline controller_interface/include/controller_interface/controller_interface_base.hpp:84]:
```suggestion
   * The claimed interfaces are populated in the
```

File: controller_interface/include/controller_interface/controller_interface_base.hpp [hunk 2]
```diff
@@ -72,9 +72,37 @@ class ControllerInterfaceBase : public rclcpp_lifecycle::node_interfaces::Lifecy
...
+  /// Get configuration for controller's required state interfaces.
+  /**
+   * Method used by controller manager to get configuration of controller's state interface.
+   * Each controller can use individual method to determine interface names that in simples case
+   * have the following format: `<joint>/<interface>`.
+   * The method is called only in `inactive` or `active` state, i.e., `on_configure` has to be
+   * called first.
+   * The configuration is used to check if controller can be activated and to claim interfaces from
+   * hardware.
+   * The claimed interface are populated in the
```
Comment (bmagyar) [inline controller_interface/include/controller_interface/controller_interface_base.hpp:101]:
```suggestion
   * The claimed interfaces are populated in the
```

File: controller_interface/include/controller_interface/controller_interface_base.hpp [hunk 3]
```diff
@@ -72,9 +72,37 @@ class ControllerInterfaceBase : public rclcpp_lifecycle::node_interfaces::Lifecy
...
+   * \ref ControllerInterfaceBase::command_interfaces_ "command_interfaces_" member.
+   *
+   * \returns configuration of command interfaces.
+   */
   CONTROLLER_INTERFACE_PUBLIC
   virtual InterfaceConfiguration command_interface_configuration() const = 0;
 
+  /// Get configuration for controller's required state interfaces.
+  /**
+   * Method used by controller manager to get configuration of controller's state interface.
```
Comment (bmagyar) [inline controller_interface/include/controller_interface/controller_interface_base.hpp:94]:
```suggestion
   * Method used by the controller_manager to get the set of state interface used by the controller.
```

File: controller_interface/include/controller_interface/controller_interface_base.hpp [hunk 4]
```diff
@@ -72,9 +72,37 @@ class ControllerInterfaceBase : public rclcpp_lifecycle::node_interfaces::Lifecy
   CONTROLLER_INTERFACE_PUBLIC
   virtual ~ControllerInterfaceBase() = default;
 
+  /// Get configuration for controller's required command interfaces.
+  /**
+   * Method used by controller manager to get configuration of controller's command interface.
```
Comment (bmagyar) [inline controller_interface/include/controller_interface/controller_interface_base.hpp:77]:
```suggestion
   * Method used by the controller_manager to get the set of command interfaces used by the controller.
```

**Review by bmagyar — COMMENTED** [review]
some touchups, feel free to merge if you are happy to apply them

**Review by bmagyar — APPROVED** [review]

**Review by bmagyar — APPROVED** [review]

**Comment by destogl** [comment]
@mergifyio backport humble

**Pull request merged**